src/main/java/org/folio/processing/mapping/defaultmapper/processor/functions/NormalizationFunction.java

```diff
@@ -543,8 +546,36 @@ public String apply(RuleExecutionContext context) {
         .findFirst()
         .orElse(STUB_FIELD_TYPE_ID);
     }
+  },
+
+  SET_AUTHORITY_SOURCE_FILE_ID() {
+    private static final String CODE_PARAMETER = "code";
+
+    @Override
+    public String apply(RuleExecutionContext context) {
+      var value = context.getRuleParameter().getString(CODE_PARAMETER);
+      var prefix = getPrefix(value);
+      var authoritySourceFiles = context.getMappingParameters().getAuthoritySourceFiles();
+
+      if (authoritySourceFiles == null || prefix == null) {
+        return null;
+      }
+
+      return authoritySourceFiles.stream()
+        .filter(authoritySourceFile -> authoritySourceFile.getCodes().stream()
+          .anyMatch(code -> code.equalsIgnoreCase(prefix)))
+        .map(AuthoritySourceFile::getId)
+        .findFirst()
+        .orElse(null);
+    }
   };
 
+  private static String getPrefix(String value) {
+    Pattern p = Pattern.compile("[a-zA-Z]+");
+    Matcher m = p.matcher(value);
+    return m.find() ? m.group() : null;
+  }
```

Comment:
This method is not searching for prefixes but also for substrings in the middle or end. For example for string `1234nmst123` it will find `nmst`